tool/scaffold/src/main/java/org/openscience/cdk/tools/scaffold/NetworkNode.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022 Julian Zander <zanderjulian@gmx.de>
+ *                    Jonas Schaub <jonas.schaub@uni-jena.de>
+ *                    Achim Zielesny <achim.zielesny@w-hs.de>
+ *                    Christoph Steinbeck <christoph.steinbeck@uni-jena.de>
+ *
+ * Contact: cdk-devel@lists.sourceforge.net
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License
+ * as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package org.openscience.cdk.tools.scaffold;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * The NetworkNodes are nodes from which a {@link ScaffoldNetwork} can be built.
+ * It is used to organise the IAtomContainers and enables a relationship between the different objects.
+ * A NetworkNode can have multiple children and parents.
+ * The parents are the smaller parent scaffolds.
+ * @param <MoleculeType> As MoleculeType, any data type can be defined.
+ *                     In our scenario, the node contains a CDK IAtomContainer.
+ *
+ * @author Julian Zander, Jonas Schaub (zanderjulian@gmx.de, jonas.schaub@uni-jena.de)
+ * @version 1.0.0.0
+ */
+public class NetworkNode <MoleculeType> extends ScaffoldNodeBase<MoleculeType> {
+
+    //<editor-fold desc="Private variables">
```

Comment:
Please remove editor suggestion lines

Reply:
Editor folds were removed with the latest commit.